hazelcast/src/main/java/com/hazelcast/version/Version.java

```diff
@@ -156,67 +162,71 @@ public static Version of(String version) {
      * @return {@code true} if this version equals {@code version}
      */
     public boolean isEqualTo(Version version) {
-        return this.equals(version);
+        return major == version.major && minor == version.minor;
     }
 
     /**
      * @param version other version to compare to
      * @return {@code true} if this version is greater than {@code version}
      */
     public boolean isGreaterThan(Version version) {
-        return this.compareTo(version) > 0;
+        return !version.isUnknown() && compareTo(version) > 0;
     }
 
     /**
      * @param version other version to compare to
      * @return {@code true} if this version is unknown or if this version is greater than {@code version}
      */
     public boolean isUnknownOrGreaterThan(Version version) {
-        return this.isUnknown() || this.compareTo(version) > 0;
+        return isUnknown() || (!version.isUnknown() && compareTo(version) > 0);
     }
 
     /**
      * @param version other version to compare to
      * @return {@code true} if this version is greater than or equal to {@code version}
      */
     public boolean isGreaterOrEqual(Version version) {
-        return this.compareTo(version) >= 0;
+        return (!version.isUnknown() && compareTo(version) >= 0) || (version.isUnknown() && isUnknown());
     }
 
     /**
      * @param version other version to compare to
      * @return {@code true} if this version is unknown or if this version is greater than or equal to {@code version}
      */
     public boolean isUnknownGreaterOrEqual(Version version) {
```

Comment:
Not related to PR, but would be great to add "Or" in the name, so `isUnknownOrGreaterOrEqual`, otherwise it reads that an unknown is greater or equal.